Omit 0-amount outputs when reconstructing CETs

luckysori opened the pull request:
This fixes a bug where we were naively adding 0-amount outputs for the party that was getting liquidated when reconstructing "liquidation" CETs.

The network would not accept these transactions, but we were failing as soon as we were checking against the original CET TXID. In `maia` we were already constructing this kind of transaction correctly, by omitting the 0-amount output.

luckysori committed:
Add more context to failed-to-decrypt-CET warning

Co-authored-by: Daniel Karzel <daniel.karzel@coblox.tech>

luckysori committed:
Omit 0-amount outputs when reconstructing CETs

This fixes a bug where we were naively adding 0-amount outputs for the
party that was getting liquidated when reconstructing "liquidation"
CETs.

The network would not accept these transactions, but we were failing
as soon as we were checking against the original CET TXID. In `maia`
we were already constructing this kind of transaction correctly, by
omitting the 0-amount output.

Co-authored-by: Daniel Karzel <daniel.karzel@coblox.tech>

da-kami reviewed (approved):
Would it be worth to add a unit test for this?

luckysori committed:
Verify validity of deconstructed CETs during debug builds

During contract setup and rollover, we deconstruct each individual CET
in order to save space. It is hard to write a test to verify that this
reconstruction logic is sound. Instead of writing a test we introduce
a `debug_assert_eq!` statement, which will only run on debug builds.

If we confirm that we can reconstruct a transaction with the original
TXID, we know that we did not mess up the deconstruction.

da-kami commented on daemon/src/setup_contract.rs:
```diff
@@ -568,20 +580,28 @@ pub async fn roll_over(
-                    })
+                    };
+
+                    debug_assert_eq!(
```
I think this is fair, but it would be interesting to have some stats because rollovers are already expensive and we would do that for *each* CET. 
Could you maybe put that into a separate PR and we merge it later and try to get some stats on it on `testnet`?
I'd like to get the fix in without this, because it is not predictable if more rollovers will fail with this change.

luckysori replied:
The change only applies to debug builds! So it doesn't affect production. I can put it in a separate PR, but I would merge it straight away 😛

da-kami reviewed (changes requested):
Please move the last commit into a separate PR!

da-kami reviewed (approved):
Sry, missed that it was a debug assert!
